test/integration/render-tests/debug/terrain/collision-pitch-with-map/style.json

```diff
@@ -3,6 +3,7 @@
   "metadata": {
     "test": {
       "height": 256,
+      "allowed": 0.0004,
```

Comment:
Instead of introducing an increased allowed baseline for every platform, how about allowing that format:
- Applies to windows only, default remains 0:
```
"allowed": {
  "windows": 0.0004,
}
```
- Applies to all platforms:
```
"allowed": 0.0004
```
- Applies a default to all platform and a windows specific one:
```
"allowed": {
  "default": 0.0001
  "windows": 0.0004
}
```

Reply:
One concern with that idea is that it could add complexity and maintenance burden, particularly as we add more browsers in the future. So it could wind up looking more like the following, which seems challenging to maintain.

```js
"allowed": {
  "default": 0.0001,
  "windows-chrome": 0.0004,
  "windows-edge": 0.0006,
  "linux-firefox": 0.0002,
  "macos-safari": 0.001
}
```

For tests where the rendering difference is severe this approach makes sense to me, but in cases where the difference is imperceptible (which is the majority of failing tests) I think that raising the default allowed a small amount creates a much more maintainable solution and would be unlikely to miss an actual regression.

Reply:
Yes that definitely makes sense to not do that for most tests, the issue with maintenance burden was also my initial thought with regards to that, so maybe we only do that as a last resort and not for every tests.

After classifying what tests fall in what failure category (I've identified 4), an approach could be:
1. The test fail on one platform, but contains debug rendering, such as:
<img width="1012" alt="Screen Shot 2022-10-20 at 1 05 19 PM" src="https://user-images.githubusercontent.com/7061573/197047021-01e45f40-627b-4d9e-a393-60af006fc1e0.png"> This category of test is usually very platform dependent due to the rendering of gl lines which are implementation specific details. In that case it's not critical to test this test on every platform, external end user won't likely benefit from this and this is most important to us to test on the developer platform (e.g. macOS). We could have an per platform ignore list for these.

2. The test fails on one platform, rendering difference is visually imperceptible by human eye or is the result of antialiasing with an `allowed` that is fairly low:
<img width="788" alt="Screen Shot 2022-10-20 at 1 09 08 PM" src="https://user-images.githubusercontent.com/7061573/197047869-5c61cce1-1bdc-4105-8481-63e1918c635c.png"><img width="790" alt="Screen Shot 2022-10-20 at 1 11 09 PM" src="https://user-images.githubusercontent.com/7061573/197048253-fa651fd5-c081-44cf-a7ec-3907874bd411.png"><img width="845" alt="Screen Shot 2022-10-20 at 1 12 15 PM" src="https://user-images.githubusercontent.com/7061573/197048566-f220de1a-c803-4a47-b760-f2628b47f31f.png"> That category of test seems minimal enough that we could increase the `allowed` globally, for every platform.

3. The test fails on one platform, rendering difference is visually imperceptible by human eye or is the result of antialiasing with an `allowed` that is too high:
<img width="1296" alt="Screen Shot 2022-10-20 at 1 14 34 PM" src="https://user-images.githubusercontent.com/7061573/197048970-b0877b78-a6a4-4b0d-8538-20441ea61f0b.png"> That category of test seems like it would fall in a per-platform `allowed` since it is too high of a baseline increase to increase globally and could be subject to unnoticed regressions.

4. Something is clearly broken and different between platforms
<img width="1026" alt="Screen Shot 2022-10-20 at 1 17 46 PM" src="https://user-images.githubusercontent.com/7061573/197049555-84cc8c1d-cd39-4143-80e3-61692140e3e8.png"> This seems like it should be put in an `ignore-windows.json` for now and investigated and addressed separately.

So overall it seems like we may want a bit of these three methods based on how we categorize our test failure:
- a per-platform ignore-list to be addressed in subsequent work when tests are clearly broken or contains non-critical render tests (such as visual debugging information)
- a per-platform `allowed` to be used very sparingly and with care to not increase maintenance burden, that is used in last resort when we want the test covered on different platforms but not increase the baseline everywhere
- a global increase of `allowed` when the difference is minimal enough so that we don't increase our tests to be as precise as the least precise test

Reply:
To keep this scope of this PR to a minimum, for now I'm adding tests like 1, 3 and 4 to the ignore list(s) and opening further tickets to address issues like 4.